### PR TITLE
CMO-240 New Relic Log number of attaches by county

### PR DESCRIPTION
**Pull request by pokharelmanish**
### Jira Story

- [Log number of attaches by county](https://osi-cwds.atlassian.net/browse/CMO-240)

## Description
Log number of attaches by county

Screen Shot 2019-10-14 at 8 45 15 AM

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/common/relationship/AttachLink.jsx [hunk 1]
```diff
@@ -37,15 +41,14 @@ export const AttachLink = ({
...
   relationship,
   screeningId,
+  county,
 }) => {
   if (
     relationship.person_card_exists &&
     !isPending(relationship, pendingPeople) &&
     !isParticipant(relationship, participants)
   ) {
-    return isScreening ?
```
Comment (pokharelmanish) [inline app/javascript/common/relationship/AttachLink.jsx]: Refactored: removed `isScreening` since it is a snapshot repo.

**Comment by pokharelmanish** [comment]
retest this please

**Comment by TylerOSI** [comment]
retest this please

**Comment by TylerOSI** [comment]
Jenkins test this please

**Comment by TylerOSI** [comment]
retest this please

File: app/javascript/common/relationship/AttachLink.jsx [hunk 2]
```diff
@@ -37,15 +41,14 @@ export const AttachLink = ({
...
-      attachLink(onClick, relationship, screeningId) :
-      attachLink(onClick, relationship)
```
[the remark this answers is not in the document]
Comment (pokharelmanish) [reply app/javascript/common/relationship/AttachLink.jsx]: Yes, if the if clause returns true it should return `attachLink(onClick, relationship)` but in our case, we also need to pass county info so `attachLink(onClick, relationship, county)`

Actually, if you see the test carefully there was a failing spec that is corrected by removing `screeningId`.

File: spec/javascripts/components/common/relationship/AttachLinkSpec.jsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import {shallow} from 'enzyme'
 import AttachLink from 'common/relationship/AttachLink'
+import * as Analytics from 'utils/analytics'
```
[the remark this answers is not in the document]
Comment (pokharelmanish) [reply spec/javascripts/components/common/relationship/AttachLinkSpec.jsx]: I am not sure about that but I was following the pattern which we did in `Autocompleter.jsx`.

[the remark this answers is not in the document]
Comment (TylerOSI) [reply spec/javascripts/components/common/relationship/AttachLinkSpec.jsx]: lets just be consistent in the code changes we make going forward (starting with this PR)

[the remark this answers is not in the document]
Comment (edgibbs) [reply spec/javascripts/components/common/relationship/AttachLinkSpec.jsx]: OK, curiosity just burned about 30 minutes on this, and apparently this is the cleanest solution, all the other options end up being a lot more boilerplate.  Turns out Jasmine doesn't handle this case well, and I've gotten to used to Jest, so it's fine.

**Pull request merged**